ding/model/common/head.py

```diff
@@ -452,6 +452,153 @@ def forward(self, x: torch.Tensor, num_quantiles: Optional[int] = None) -> Dict:
+            - norm_type (:obj:`str`):
+                The type of normalization to use, see ``ding.torch_utils.fc_block`` for more details
+            - noise (:obj:`bool`): Whether use noisy ``fc_block``
+        """
+        super(FQFHead, self).__init__()
+        layer = NoiseLinearLayer if noise else nn.Linear
+        block = noise_block if noise else fc_block
+        self.Q = nn.Sequential(
+            MLP(
+                hidden_size,
+                hidden_size,
+                hidden_size,
+                layer_num,
+                layer_fn=layer,
+                activation=activation,
+                norm_type=norm_type
+            ), block(hidden_size, output_size)
+        )
+        self.num_quantiles = num_quantiles
+        self.quantile_embedding_size = quantile_embedding_size
+        self.output_size = output_size
+        self.iqn_fc = nn.Linear(self.quantile_embedding_size, hidden_size)
+        #self.beta_function = beta_function_map[beta_function_type]
+        self.quantiles_proposal = nn.Sequential(
+            MLP(
+                hidden_size,
+                hidden_size,
+                num_quantiles,
+                layer_num = 1,
+                layer_fn=nn.Linear,
+                activation=nn.Softmax(),
+                norm_type=norm_type
+            )
+        )
+    # def quantiles_proposal(self, x: torch.Tensor, num_quantiles: Optional[int] = None) -> torch.Tensor:
+    #   return
+
+    def quantile_net(self, quantiles: torch.Tensor) -> torch.Tensor:
+        r"""
+        Overview:
+           Deterministic parametric function trained to reparameterize samples from a base distribution.
+           By repeated Bellman update iterations of Q-learning, the optimal action-value function is estimated.
+        Arguments:
+            - x (:obj:`torch.Tensor`): The encoded embedding tensor of parametric sample
+        Returns:
+            - (:obj:`torch.Tensor`):
+                QN output tensor after reparameterization of shape ``(quantile_embedding_size, output_size)``
+        Examples:
+            >>> head = QuantileHead(64, 64)
+            >>> quantiles = torch.randn(4,32)
+            >>> qn_output = head.quantile_net(quantiles)
+            >>> assert isinstance(qn_output, torch.Tensor)
+            >>> # default quantile_embedding_size: int = 128,
+            >>> assert qn_output.shape == torch.Size([128, 64])
+        """
+        batch_size = quantiles.shape[0]
+        num_quantiles = quantiles.shape[1]
+        quantile_net = torch.cos(
+            torch.arange(1, self.quantile_embedding_size + 1, 1).view(1, 1, self.quantile_embedding_size).to(quantiles) * math.pi * quantiles.view(batch_size, num_quantiles, 1)
+        ).view(batch_size * num_quantiles, self.quantile_embedding_size)
+        quantile_net = self.iqn_fc(quantile_net)
+        quantile_net = F.relu(quantile_net).view(batch_size, num_quantiles, -1)   #(batch,num_quantiles,hidden_size)
```

Comment:
move this relu to `nn.ReLU()` in `__init__` method so that we can know the structure of model by calling `print(model)`